### ISSUE TITLE
`evar_{,filtered}_env` considered harmful.

### ISSUE THREAD

**Issue by ejgallego** [coq/coq#6523]
Both `Evd.evar_env` and `Evd.evar_filtered_env` are implemented using a call to `Global.env ()`, their current form should eventually go away.

**Comment by ejgallego** [comment]
IMHO the full fix for this is more involved, eventually these functions should not exist at all.

**Comment by ppedrot** [comment]
I don't think that this issue is relevant anymore. We're removed the environment and pushing the global calls upwards, but it's hardly related to this precise API. I'm closing it because I don't see how we would fix it more than it is currently.
